backend/api/simulation.py

```
@@ -1469,8 +1885,51 @@ def _safe_scene_id():
+                    persona_id = target_persona.get('db_id')
+                    custom_prompt = target_persona.get('system_prompt')
+                    # DB verification (read-only): log what's actually stored for this persona's system_prompt
+                    try:
+                        from database.models import ScenarioPersona as _DBPersona
+                        db_check = db.query(_DBPersona).filter(_DBPersona.id == persona_id).first()
+                        if db_check is not None:
+                            db_prompt = db_check.system_prompt
+                            has_db_prompt = isinstance(db_prompt, str) and db_prompt.strip() != ""
+                            # logging removed
+                            # If DB has a prompt and orchestrator persona lacks it or differs, sync and use DB value verbatim
+                            if has_db_prompt and (not has_custom or (isinstance(custom_prompt, str) and custom_prompt.strip() != db_prompt.strip())):
```

Ruff (0.14.0)

Undefined name `has_custom`

(F821)